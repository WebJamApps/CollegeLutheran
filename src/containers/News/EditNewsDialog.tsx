import {
  Button,
  Checkbox,
  Dialog, DialogActions, DialogContent, DialogTitle, FormControlLabel, FormGroup, TextField,
} from '@mui/material';
import { useContext } from 'react';
import { AuthContext } from 'src/providers/Auth.provider';
import { ContentContext } from 'src/providers/Content.provider';
import utils, { defaultNews } from './utilsN';

interface InewsTextFieldProps {
  value: string,
  label: string,
  className: string
  onChange: (arg0: any) => any
}

export function NewsTextField(props: InewsTextFieldProps) {
  const {
    value, label, onChange, className,
  } = props;
  return (
    <TextField
      sx={{ marginTop: '20px' }}
      label={label}
      type="text"
      fullWidth
      className={className}
      value={value}
      onChange={onChange}
    />
  );
}

function checkDisabled(editNews: typeof defaultNews): boolean {
  return !!(editNews.title && editNews.url);
}

interface IeditNewsContentProps {
  editNewsState:IeditNewsDialogProps, showHideCaption:(evt: any) => void
}
export function EditNewsContent(props:IeditNewsContentProps) {
  const { editNewsState, showHideCaption } = props;
  const { editNews, setEditNews } = editNewsState;
  return (
    <DialogContent sx={{ padding: '10px 10px' }}>
      <NewsTextField
        value={editNews.url as string}
        label="* URL"
        className="url"
        onChange={(evt) => {
          const { target: { value } } = evt;
          setEditNews({ ...editNews, url: value });
          return value;
        }}
      />
      <NewsTextField
        value={editNews.title}
        label="* Title"
        className="title"
        onChange={(evt) => {
          const { target: { value } } = evt;
          setEditNews({ ...editNews, title: value });
          return value;
        }}
      />
      <FormGroup>
        <FormControlLabel
          control={(
            <Checkbox
              checked={editNews.comments === 'worshipbulletin'}
              onClick={showHideCaption}
            />
      )}
          label="Show as Worship Bulletin"
        />
      </FormGroup>
    </DialogContent>
  );
}

export interface IeditNewsDialogProps {
  editNews: typeof defaultNews, setEditNews: (arg0: typeof defaultNews) => void,
}

export function EditNewsButtons(props: IeditNewsDialogProps) {
  const { editNews, setEditNews } = props;
  const { auth } = useContext(AuthContext);
  const { getNews } = useContext(ContentContext);
  return (
    <DialogActions>
      <>
        <Button
          disabled={!checkDisabled(editNews)}
          size="small"
          variant="contained"
          className="updateNewsButton"
              // eslint-disable-next-line @typescript-eslint/no-floating-promises
          onClick={() => { utils.updateNews(editNews, auth, getNews, setEditNews); }}
        >
          Update
        </Button>
        <Button
          style={{ backgroundColor: 'red', color: 'white' }}
          size="small"
          className="deleteNewsButton"
              // eslint-disable-next-line @typescript-eslint/no-floating-promises
          onClick={() => { utils.deleteNews(editNews, getNews, auth, setEditNews); }}
        >
          Delete
        </Button>
        <Button
          size="small"
          className="cancelNewsButton"
          onClick={() => setEditNews(defaultNews)}
        >
          Cancel
        </Button>
      </>
    </DialogActions>
  );
}

export function EditNewsDialog({ editNews, setEditNews }: IeditNewsDialogProps) {
  const showHideCaption = utils.makeShowHideBulletin(setEditNews, editNews);
  return (
    <Dialog
      disableEnforceFocus
      disableAutoFocus
      className="editNewsDialog"
      open={editNews._id !== ''}
      onClose={() => setEditNews(defaultNews)}
    >
      <DialogTitle>Edit News</DialogTitle>
      <EditNewsContent showHideCaption={showHideCaption} editNewsState={{ editNews, setEditNews }} />
      <EditNewsButtons editNews={editNews} setEditNews={setEditNews} />
    </Dialog>
  );
}

