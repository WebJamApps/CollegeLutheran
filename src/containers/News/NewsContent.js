import React from 'react';
import PropTypes from 'prop-types';
import ELCALogo from '../../components/elcaLogo';

const pageName = 'news';

const NewsContent = ({ books }) => (
  <div className="page-content">
    <div className="container-fluid">
      <p style={{ fontSize: '4pt', margin: '0' }}>&nbsp;</p>
      <div
        className="news-div"
        style={{
          margin: 'auto',
          textAlign: 'center',
          padding: '0',
          marginLeft: '-8px',
        }}
      >
        <iframe
          style={{
            width: '100%',
            height: '540px',
            border: '1px solid #d3d3d3',
          }}
          className="newsIFrame"
          src="https://visitor.r20.constantcontact.com/d.jsp?
            llr=9mgiltbab&p=oi&m=1101234706842&sit=ah8uvihbb&f=f26b8fbd-329c-49ba-b2c2-68eee26dbef1"
          title="News Content"
        >
          <p>Your browser does not support iframe.</p>
        </iframe>
      </div>
      <div>
        {books.length > 0 ? (
          <div className="forumsTable">
            <h3
              style={{
                paddingTop: '22px',
                marginBottom: '8px',
                fontWeight: '100',
                textAlign: 'center',
              }}
            >
              Monthly Forums
            </h3>
            <div className="table-responsive">
              <table className="newsTable">
                <thead>
                  <tr>
                    <th style={{ padding: '8px', width: '230px' }}>Title</th>
                    <th style={{ padding: '8px', width: '105px' }}>Date</th>
                  </tr>
                </thead>
                <tbody>
                  {books.map((d) => (
                    <tr key={d._id}>
                      <td className="newsUrl">
                        <a rel="noopener noreferrer" target="_blank" href={d.url}>{d.title}</a>
                      </td>
                      <td>{d.created_at.split('T')[0]}</td>
                    </tr>
                  ))}
                </tbody>
              </table>
            </div>
          </div>
        ) : (
          <p>There are currently no pdfs links to display</p>
        )}
      </div>
    </div>
    <p>&nbsp;</p>
    <p>&nbsp;</p>
    <ELCALogo pageName={pageName} />
  </div>
);

NewsContent.defaultProps = { books: [] };
NewsContent.propTypes = {
  books: PropTypes.arrayOf(
    PropTypes.shape({
      comments: PropTypes.string,
    }),
  ),
};

export default NewsContent;
