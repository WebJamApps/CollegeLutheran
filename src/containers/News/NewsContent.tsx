import React from 'react';
import ELCALogo from '../../components/elcaLogo';
import { Ibook } from '../../redux/mapStoreToProps';

interface NewsContentProps {
  books?: Ibook[];
}

const pageName = 'news';

const NewsContent = ({ books }: NewsContentProps): JSX.Element => (
  <div className="page-content">
    <div>
      <div>
        {books && books.length > 0 ? (
          <div className="forumsTable">
            <h3
              style={{
                paddingTop: '22px',
                marginBottom: '8px',
                textAlign: 'center',
              }}
            >
              Announcements
            </h3>
            <div className="TableStyle-root">
              <div className="TableContainer-root table-responsive">
                <table className="TableHeader newsTable" aria-label="News Anouncement Table">
                  <thead className="TableHead-root">
                    <tr className="TableRow-root">
                      <th className="TableCell-root TableCell-head StickyHeader newsTitle">Title</th>
                      <th className="TableCell-root TableCell-head StickyHeader" style={{ width: '103px' }}>Date</th>
                    </tr>
                  </thead>
                  <tbody className="TableBody-root">
                    {books && books.map(
                      (d: { _id: string | number | undefined; url?: string | undefined; title: React.ReactNode; created_at: string; }) => (
                        <tr className="TableRow-root" key={d._id}>
                          <td className="TableCell-root TableCell-body newsUrl">
                            <a rel="noopener noreferrer" target="_blank" href={d.url}>{d.title}</a>
                          </td>
                          <td className="TableCell-root TableCell-body">{d.created_at.split('T')[0]}</td>
                        </tr>
                      ),
                    )}
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        ) : (
          <p>There are currently no pdfs links to display</p>
        )}
      </div>
      <p style={{ fontSize: '4pt', margin: '0' }}>&nbsp;</p>
      <hr style={{ margin: '0px' }} />
      <div
        className="news-div"
        style={{
          margin: '0px 0px auto -8px',
          textAlign: 'center',
          padding: '0',
          width: '108%',
          overflow: 'hidden',
        }}
      >
        <iframe
          className="newsIFrame"
          style={{
            overflow: 'hidden',
            padding: 0,
            width: '100%',
            height: '540px',
            border: 'none',
          }}
          src="https://visitor.r20.constantcontact.com/d.jsp?
            llr=9mgiltbab&p=oi&m=1101234706842&sit=ah8uvihbb&f=f26b8fbd-329c-49ba-b2c2-68eee26dbef1"
          title="News Content"
        >
          <p>Your browser does not support iframe.</p>
        </iframe>
      </div>
    </div>
    <ELCALogo pageName={pageName} />
  </div>
);

NewsContent.defaultProps = { books: [] };

export default NewsContent;
