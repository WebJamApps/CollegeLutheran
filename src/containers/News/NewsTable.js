import React, { Component } from 'react';
import MUIDataTable from 'mui-datatables';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
import ReactHtmlParser from 'react-html-parser';
import PropTypes from 'prop-types';
import 'core-js/stable';
import 'regenerator-runtime/runtime';
import { connect } from 'react-redux';
import mapStoreToProps from '../../redux/mapStoreToProps';

export class NewsTable extends Component {
  constructor(props) {
    super(props);
    this.setColumns = this.setColumns.bind(this);
    this.getMuiTheme = this.getMuiTheme.bind(this);
    this.checkNewsTable = this.checkNewsTable.bind(this);
    this.setColumns = this.setColumns.bind(this);
    this.state = {
      columns: [],
    };
  }

  componentDidMount() { this.setColumns(); }

  componentDidUpdate(prevProps) {
    const { newsUpdated } = this.props;
    return this.checkNewsTable(prevProps.newsUpdated, newsUpdated);
  }

  getMuiTheme() { // eslint-disable-line class-methods-use-this
    return createMuiTheme({
      typography: {
        useNextVariants: true,
      },
      overrides: {
        MUIDataTableHeadCell: {
          root: {
            padding: '4px', fontWeight: 'bold', color: 'black', fontSize: '11pt',
          },
        },
        MuiTableRow: { head: { height: '40px' } },
        MuiTableCell: { root: { padding: '4px' } },
        MUIDataTableToolbar: { actions: { display: 'none' }, root: { paddingLeft: 0, minHeight: 'inherit' } },
        MUIDataTable: { responsiveScroll: { maxHeight: '4.3in' } },
        MuiTypography: { h6: { color: 'black', fontWeight: 'bold', fontStyle: 'italic' } },
      },
    });
  }

  setColumns() {
    const columns = [];
    const titles = ['Date', 'Time', 'Location', 'Venue', 'Tickets'];
    for (let i = 0; i < titles.length; i += 1) {
      columns.push({
        name: titles[i].toLowerCase(), // eslint-disable-line security/detect-object-injection
        label: titles[i], // eslint-disable-line security/detect-object-injection
        options: {
          filter: false,
          sort: false,
          customBodyRender: (value) => (
            <div style={{ minWidth: '1.3in' }}>
              <p style={{ margin: 0, fontSize: 'inherit' }}>
                { ReactHtmlParser(value) }
              </p>
            </div>
          ),
        },
      });
    }
    this.setState({ columns });
  }

  checkNewsTable(pTupdated, nTupdated) {
    if (!pTupdated && nTupdated) {
      const { dispatch } = this.props;
      dispatch({ type: 'RESET_NEWS' });
      this.setState({ columns: [] });
      this.setColumns();
      return Promise.resolve(true);
    }
    return Promise.resolve(false);
  }

  render() {
    const { columns } = this.state;
    const { news } = this.props;
    return (
      <div className="newsTable">
        <div style={{ maxWidth: '100%' }}>
          <MuiThemeProvider theme={this.getMuiTheme()}>
            <MUIDataTable
              options={{
                filterType: 'dropdown',
                pagination: false,
                responsive: 'scrollMaxHeight',
                filter: false,
                download: false,
                search: false,
                print: false,
                viewColumns: false,
                selectableRows: 'none',
                fixedHeader: false,
              }}
              columns={columns}
              data={news}
              title="News"
            />
          </MuiThemeProvider>
        </div>
      </div>
    );
  }
}
NewsTable.propTypes = {
  dispatch: PropTypes.func.isRequired,
  newsUpdated: PropTypes.bool.isRequired,
  news: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

export default connect(mapStoreToProps)(NewsTable);
