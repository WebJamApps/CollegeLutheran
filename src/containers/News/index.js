import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DefaultNewsContent from './NewsContent';
import mapStoreToProps from '../../redux/mapStoreToProps';
import commonUtils from '../../lib/commonUtils';

export const News = ({ books }) => {
  commonUtils.setTitleAndScroll('News & Forum');
  books.sort((a, b) => {
    const dataA = a.created_at.split('T')[0];
    const dateB = b.created_at.split('T')[0];
    if (dataA < dateB) return 1;
    if (dataA > dateB) return -1;
    return 0;
  });
  return <DefaultNewsContent books={books} />;
};

News.defaultProps = { books: [] };
News.propTypes = {
  books: PropTypes.arrayOf(
    PropTypes.shape({
      comments: PropTypes.string,
    }),
  ),
};

export default connect(mapStoreToProps, null)(News);
