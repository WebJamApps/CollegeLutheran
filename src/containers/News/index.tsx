import React from 'react';
import { connect } from 'react-redux';
import DefaultNewsContent from './NewsContent';
import mapStoreToProps, { Ibook } from '../../redux/mapStoreToProps';
import commonUtils from '../../lib/commonUtils';

export function News(props: { books: Ibook[] }): JSX.Element {
  const { books } = props;
  commonUtils.setTitleAndScroll('News', window.screen.width);
  books.sort((a, b) => {
    const dataA = a.created_at.split('T')[0];
    const dateB = b.created_at.split('T')[0];
    if (dataA < dateB) return 1;
    if (dataA > dateB) return -1;
    return 0;
  });
  return <DefaultNewsContent books={books} />;
}

export default connect(mapStoreToProps, null)(News);
