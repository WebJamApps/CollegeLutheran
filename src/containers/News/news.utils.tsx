import axios from 'axios';
import commonUtils from 'src/lib/commonUtils';
import { Iauth } from 'src/providers/Auth.provider';
import { Ibook } from 'src/providers/utils';

export const defaultNews = {
  url: '', comments: '', title: '', _id: '', type: 'Forum',
} as Ibook;

async function updateNews(
  editNews: typeof defaultNews,
  auth: Iauth,
  getNews: () => Promise<void>,
  setEditNews: (arg0:typeof defaultNews) => void,
  setIsSubmitting: (arg0:boolean) => void,
): Promise<void> {
  try {
    setIsSubmitting(true);
    const { token } = auth;
    const config = {
      url: `${process.env.BackendUrl}/book`,
      method: 'put',
      headers: { Authorization: `Bearer ${auth.token}`, Accept: 'application/json' },
      data: editNews,
    };
    // const { status } = await axios.put(config.url, { editNews, token });
    const { status } = await axios.request(config);
    if (status === 200) {
      await commonUtils.delay(2);
      setIsSubmitting(false);
      setEditNews(defaultNews);
      await getNews();
    }
  } catch (err) { console.log((err as Error).message); }
}

async function deleteNews(
  getNews: () => Promise<void>,
  auth: Iauth,
  setEditNews: (arg0: typeof defaultNews) => void,
  setIsSubmitting: (arg0: boolean) => void,
): Promise<void> {
  try {
    setIsSubmitting(true);
    const config = {
      url: `${process.env.BackendUrl}/book`,
      method: 'put',
      headers: { Authorization: `Bearer ${auth.token}`, Accept: 'application/json' },
    };
    const { status } = await axios.delete(config.url);
    if (status === 200) {
      await commonUtils.delay(2);
      setIsSubmitting(false);
      setEditNews(defaultNews);
      await getNews();
    }
  } catch (err) { console.log((err as Error).message); }
}

const makeShowHideCaption = (setNews: (arg0: typeof defaultNews) => void, news: typeof defaultNews) => (evt: any) => {
  const { target: { checked } } = evt;
  const comments = checked ? 'worshipbulletin' : '';
  setNews({ ...news, comments });
};

export default { updateNews, deleteNews, makeShowHideCaption };
