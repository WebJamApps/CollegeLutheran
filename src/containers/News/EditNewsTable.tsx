import { Button } from '@mui/material';
import {
  GridColDef, GridRowsProp,
  GridRenderCellParams, DataGrid,
} from '@mui/x-data-grid';
import { useContext, useState } from 'react';
import { ContentContext } from 'src/providers/Content.provider';
import { defaultNews } from './utilsN';
import { EditNewsDialog } from './EditNewsDialog';

export const columns: GridColDef[] = [
  {
    field: 'title',
    headerName: 'Title',
    minWidth: 300,
    flex: 1,
    editable: false,
  },
  {
    field: 'url',
    headerName: 'link',
    width: 300,
    editable: false,
    renderCell: (params: GridRenderCellParams) => {
      const { row: { url } } = params;
      if (!url) return '';
      return url;
    },
  },
];

interface IeditNewsTableProps {
  setShowTable: (arg0:boolean) => void
}

export function EditNewsTable(props:IeditNewsTableProps) {
  const { setShowTable } = props;
  const { news } = useContext(ContentContext);
  const [editNews, setEditNews] = useState(defaultNews);
  const rows: GridRowsProp = [
    { news },
  ];
  return (
    <div
      className="editNewsTable"
      style={{
        display: 'inlineBlock', margin: 'auto', textAlign: 'center', paddingTop: '10px', height: '60vh',
      }}
    >
      <span>Select Which Link to Edit</span>
      <Button
        sx={{ marginLeft: '10px' }}
        size="small"
        variant="outlined"
        className="cancelEditNewsButton"
        onClick={() => setShowTable(false)}
      >
        Cancel

      </Button>
      <DataGrid
        onRowClick={(rowParams) => {
          setEditNews(rowParams.row);
        }}
        rows={rows}
        columns={columns}
      />
      <EditNewsDialog
        editNews={editNews}
        setEditNews={setEditNews}
        setShowTable={setShowTable}
      />
    </div>
  );
}
