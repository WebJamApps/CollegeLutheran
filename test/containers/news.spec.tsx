/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { News } from '../../src/containers/News/index';
import DefaultNewsContent from '../../src/containers/News/NewsContent';

describe('News', () => {
  const booksArr: any[] = [{
    _id: '1', created_at: '1971-01-03T00:00:00Z', title: 'howdy', type: '',
  },
  {
    _id: '3', created_at: '1973-01-03T00:00:00Z', title: 'howdy', type: '',
  },
  {
    title: 'howdy2', _id: '2', created_at: '1972-01-03T00:00:00Z', type: '',
  }];
  const controller = {
    setState: () => true,
    props: {},
    state: {
      books: booksArr,
    },
  };

  it('Renders the News component', () => {
    const results = shallow(<News books={[]} />);
    expect(results.find(DefaultNewsContent)
      .dive()
      .find('div.page-content')
      .exists()).toBe(true);
  });
  it('Renders the News component correctly', () => {
    const tree = renderer
      .create(<News books={[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the table when books exist', () => {
    const { books } = controller.state;
    const key = '1';
    const results = shallow(<News key={key} books={books} />);
    expect(results.find(DefaultNewsContent)
      .dive()
      .find('div.forumsTable')
      .exists()).toBe(true);
  });
  it('sorts the books in default order', () => {
    const bArr: any[] = [{
      title: 'howdy2', _id: '1', created_at: '1972-01-03T00:00:00Z', type: '',
    },
    {
      title: 'howdy3', _id: '2', created_at: '1972-01-03T00:00:00Z', type: '',
    }];
    const controller2 = {
      setState: () => true,
      props: {},
      state: {
        books: bArr,
      },
    };
    const { books } = controller2.state;
    const results = shallow(<News books={books} />);
    books.sort((a, b) => {
      const dataA = a.created_at;
      const dateB = b.created_at;
      if (dataA < dateB) return -1;
      return 0;
    });
    expect(results).toBeTruthy();
  });
});
