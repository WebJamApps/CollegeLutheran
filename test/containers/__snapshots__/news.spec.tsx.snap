// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Renders the News component correctly 1`] = `
<div
  className="page-content"
>
  <div>
    <div>
      <p>
        There are currently no pdfs links to display
      </p>
    </div>
    <p
      style={
        Object {
          "fontSize": "4pt",
          "margin": "0",
        }
      }
    >
      Â 
    </p>
    <hr
      style={
        Object {
          "margin": "0px",
        }
      }
    />
    <div
      className="news-div"
      style={
        Object {
          "margin": "0px 0px auto -8px",
          "overflow": "hidden",
          "padding": "0",
          "textAlign": "center",
          "width": "108%",
        }
      }
    >
      <iframe
        className="newsIFrame"
        src="https://visitor.r20.constantcontact.com/d.jsp?
            llr=9mgiltbab&p=oi&m=1101234706842&sit=ah8uvihbb&f=f26b8fbd-329c-49ba-b2c2-68eee26dbef1"
        style={
          Object {
            "border": "none",
            "height": "540px",
            "overflow": "hidden",
            "padding": 0,
            "width": "100%",
          }
        }
        title="News Content"
      >
        <p>
          Your browser does not support iframe.
        </p>
      </iframe>
    </div>
  </div>
  <div
    className="newsELCA"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <a
      href="http://www.elca.org/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="ELCA LOGO"
        id="elcaLogo"
        src="https://dl.dropboxusercontent.com/s/wkzubcmmm3pqst4/elca-logo.png?dl=0"
        style={
          Object {
            "margin": "auto",
            "marginLeft": "-12px",
            "paddingTop": "30px",
            "width": "310px",
          }
        }
      />
    </a>
  </div>
</div>
`;
